src/hooks/users.ts

```diff
@@ -1,52 +1,61 @@
 import { useMemo } from "react";
-import { useFirestoreConnect } from "react-redux-firebase";
+import { isLoaded } from "react-redux-firebase";
 
 import { usersSelector, usersByIdSelector } from "utils/selectors";
 import { WithId } from "utils/id";
 
 import { useSelector } from "hooks/useSelector";
 import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
 import { useVenueId } from "hooks/useVenueId";
+import { useSparkleFirestoreConnect } from "hooks/useSparkleFirestoreConnect";
 
 import { User } from "types/User";
 
 const useConnectUsers = () => {
-  useFirestoreConnect("users");
+  const venueId = useVenueId();
+
+  useSparkleFirestoreConnect([
+    {
+      collection: "users",
+      where: ["enteredVenueIds", "array-contains", venueId],
+      storeAs: "users",
+    },
+  ]);
```

Comment:
With this new `where` filter, are we confident that `enteredVenueIds` will have been updated with the `venueId` correctly for the current user before any code that relies on this data runs? Is there a chance that we could have a race condition that might break something based on this?

Reply:
Not sure about what I can do about that so I ensured that `venueId` is not undefined